pandas/tests/frame/test_timeseries.py

```diff
@@ -323,6 +323,12 @@ def test_asfreq_datetimeindex(self):
         ts = df['A'].asfreq('B')
         tm.assertIsInstance(ts.index, DatetimeIndex)
 
+    def test_asfreq_datetimeindex_empty_series(self):
+        empty_series = Series(index=pd.DatetimeIndex([]), data=[]).asfreq('H')
+        normal_series = Series(index=pd.DatetimeIndex(["2016-09-29 11:00"]),
+                                                      data=[3]).asfreq('H')
+        self.assertEqual(empty_series.index.freq, normal_series.index.freq)
```

Comment:
@jreback But we actually want to compare the frequency of the index not the complete series. Also, if we have one empty and one non-empty series, how can we use `assert_series_equal` method?